tests/manager/manager-ctor-auth_mechanism-001.phpt

```diff
@@ -7,9 +7,11 @@ $tests = [
     ['mongodb://username@127.0.0.1/?authMechanism=MONGODB-X509', []],
     ['mongodb://127.0.0.1/?authMechanism=MONGODB-X509', []],
     ['mongodb://username@127.0.0.1/?authMechanism=GSSAPI', []],
+    ['mongodb://127.0.0.1/?authMechanism=MONGODB-AWS', []],
```

Comment:
This is just testing that the mechanism is not rejected by libmongoc, correct?

Does acceptance of this mechanism depend on whether `MONGOC_ENABLE_MONGODB_AWS_AUTH` is enabled at compile time, or is it similar to GSSAPI as I describe below?

I don't think we have any conditional builds of the driver without SASL, but it looks like GSSAPI support is conditional on `MONGOC_ENABLE_SASL` being defined at compile time. Perhaps that only extends to the authentication attempt at runtime and not URI option validation, though, since `_mongoc_cluster_auth_node_sasl` still gets defined when SASL is not available and simply raises an error.

If the AWS mechanism behaves similarly, this LGTM. I assume we'll need more specific checks in the PHPLIB integration tests to skip if the mechanism (and KMS message) is not available.

Reply:
> This is just testing that the mechanism is not rejected by libmongoc, correct?

Yes, that's correct. Since libmongoc handles authentication for us, we don't even have a way of checking whether it picks the right credentials, as is necessary for the auth spec tests. Thus, the test is only there to make sure that we
a) don't reject the auth mechanism
b) accept the auth mechanism without a username

> Does acceptance of this mechanism depend on whether `MONGOC_ENABLE_MONGODB_AWS_AUTH` is enabled at compile time, or is it similar to GSSAPI as I describe below?
> 
> I don't think we have any conditional builds of the driver without SASL, but it looks like GSSAPI support is conditional on `MONGOC_ENABLE_SASL` being defined at compile time. Perhaps that only extends to the authentication attempt at runtime and not URI option validation, though, since `_mongoc_cluster_auth_node_sasl` still gets defined when SASL is not available and simply raises an error.

I'll test this on a special build without MONGODB-AWS to see how it behaves and will adjust accordingly. Thanks for bringing it up 👍

> If the AWS mechanism behaves similarly, this LGTM. I assume we'll need more specific checks in the PHPLIB integration tests to skip if the mechanism (and KMS message) is not available.

I'm not sure we can add a lot of integration tests: we can only blackbox-test authentication against a MONGODB-AWS system, and I have yet to check how much effort it is to set it up. Given how we're not really involved in the auth process I'd rather only test that we handle the arguments appropriately and let the libmongoc test suite take care of the rest.